clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -534,13 +535,13 @@ public ServerGroup getServerGroup(
     if (serverGroupName == null) {
       throw new Error("Invalid Server Group");
     }
-    // TODO - remove the application filter.
-    String application = StringUtils.substringBefore(serverGroupName, "-");
+
     Map<String, Set<EcsServerCluster>> clusterMap = new HashMap<>();
 
     try {
       AmazonCredentials credentials = getEcsCredentials(account);
-      clusterMap = findClusters(clusterMap, credentials, application);
+      // Can't filter by application as there's not enough information in the serverGroupName
+      clusterMap = findClusters(clusterMap, credentials);
```

Comment:
Would this be a regression?

Reply:
It shouldn't be as the `clusterMap` collection is still filtered to find the matching `serverGroupName` a little further down.

https://github.com/spinnaker/clouddriver/pull/4936/files/4d5938d458e864ffaadcf14223f3195a91c5e3e4#diff-08ef4737fffbc8ef45383c06e16b3a2f1a9a2fef2d4b4b314665b4f134c0b5a4R553

Given that the server group name still includes the application name this should still meet the requirements of the interface.